### ISSUE TITLE
Not installable with latest ppx_deriving

### ISSUE THREAD

**Issue by hcarty** [janestreet/ppx_type_conv#4]
ppx_deriving 4.0 fixes compatibility with OCaml 4.03.0 but it conflicts with ppx_type_conv in opam.  Can a new release be made to fix this incompatibility?  Among other things, it prevents cohttp and ppx_deriving from being installed under the same 4.03.0 switch.

**Comment by yminsky** [comment]
Is it clear if this is an issue in ppx_type_conv or ppx_deriving?

**Comment by hcarty** [comment]
ppx_deriving 4.0 has an incompatible interface change versus 3.x. My understanding is that there is some adaptation required on the part of ppx_type_conv to adapt to the 4.x API. I haven't looked into the required changes in detail.

**Comment by trefis** [comment]
This is the same issue as #1 and should be fixed by https://github.com/ocaml/opam-repository/pull/6861 .

For our next (4.03-compatible) release we will try and update our ppx_deriving support to the latest version.

**Comment by hcarty** [comment]
Similar at least to #1 - this issue asking about a fix for the incompatibility between ppx_type_conv and ppx_deriving. I look forward to the updated release!

**Comment by hcarty** [comment]
FWIW, it looks like this is not limited to 4.03.0.

**Comment by little-arhat** [comment]
@trefis hello, thanks for your work on this library! Any good news on compatibility with 4.03 and latest ppx_deriving?

/cc @fxfactorial  (little-arhat/ppx_jsobject_conv#1)

**Comment by ghost** [comment]
Basically we have updated almost all our ppx rewriters to support inline records. There is only one still being reviewed. Once this is done we'll do a new major release of our packages properly supporting 4.03. This should happen next week

**Comment by kitfre** [comment]
Would this also cause the following code to error on 4.03.0? 
`type bst = Leaf | Branch of bst * int * bst [@@deriving show]`
When I run this I get the following error:
`ppx_type_conv 'show' is not a supported type type-conv generator.`

**Comment by ghost** [comment]
I can't think of a reason why, feel free to open a new ticket about it

**Comment by little-arhat** [comment]
Hello, any progress on compatibility with latest ppx_deriving ? ppx_deriving_yojson wants ppx_deriving > 4.0 on 4.03, for example.

**Comment by fxfactorial** [comment]
FWIW, I don't need this anymore for my original intention of reactjs bindings (having removed them) but I do get hit with this issue in my other project(Which is way more important), `usbmux` since I have these depends: 

``` shell
depends: [
  "base-threads"
  "cmdliner" {build & >= "0.9.8"}
  "cohttp" {>= "0.21.0"}
  "lwt" {>= "2.5.2"}
  "oasis" {build & >= "0.4"}
  "ocamlbuild" {build}
  "ocamlfind" {build}
  "plist"
  "ppx_deriving" {>= "4.0"}
  "ppx_deriving_yojson" {>= "3.0"}
  "yojson" {>= "1.3.2"}
]
```

and so can't use this on `4.03.0`

**Comment by soupault** [comment]
Are there any news on this matter?

```
[egor@host ~]$ opam install core_bench async core_kernel
The following actions will be performed:
  ↘  downgrade ppx_deriving      4.0 to 3.3      [required by ppx_type_conv]
  ∗  install   ppx_type_conv     113.33.01+4.03  [required by ppx_assert]
  ∗  install   ppx_variants_conv 113.33.00+4.03  [required by ppx_expect, ppx_jane]
  ∗  install   ppx_typerep_conv  113.33.00+4.03  [required by ppx_jane]
  ∗  install   ppx_sexp_conv     113.33.01+4.03  [required by ppx_expect, ppx_assert, ppx_jane]
  ∗  install   ppx_fields_conv   113.33.00+4.03  [required by ppx_expect, ppx_jane]
  ∗  install   ppx_enumerate     113.33.00+4.03  [required by ppx_jane]
  ∗  install   ppx_compare       113.33.00+4.03  [required by ppx_expect, ppx_assert, ppx_jane]
  ∗  install   ppx_bin_prot      113.33.00+4.03  [required by ppx_jane]
  ∗  install   ppx_sexp_value    113.33.00+4.03  [required by ppx_jane]
  ∗  install   ppx_sexp_message  113.33.00+4.03  [required by ppx_jane]
  ∗  install   ppx_custom_printf 113.33.00+4.03  [required by ppx_expect, ppx_jane]
  ∗  install   ppx_assert        113.33.00       [required by core_bench, async, core_kernel]
  ∗  install   ppx_expect        113.33.01+4.03  [required by core_bench, async, core_kernel]
  ∗  install   ppx_jane          113.33.00       [required by core_bench, async, core_kernel]
  ∗  install   core_kernel       113.33.01+4.03
  ∗  install   re2               113.33.00+4.03  [required by core_extended]
  ∗  install   core              113.33.02+4.03  [required by async, core_bench]
  ∗  install   async_kernel      113.33.00       [required by async]
  ∗  install   textutils         113.33.00       [required by core_bench]
  ∗  install   async_unix        113.33.00+4.03  [required by async]
  ∗  install   async_rpc_kernel  113.33.00       [required by async_extra]
  ∗  install   core_extended     113.33.00+4.03  [required by core_bench]
  ∗  install   async_extra       113.33.00+4.03  [required by async]
  ∗  install   core_bench        113.33.00+4.03
  ∗  install   async             113.33.00     
  ↻  recompile cstruct           2.3.0           [uses async]
===== ∗  25   ↻  1   ↘  1 =====
```

**Comment by ghost** [comment]
I imported the same fix as in the master branch and did a new minor release: ocaml/opam-repository#7368

**Comment by ghost** [comment]
Minor release with the fix has been integrated in opam

**Comment by hcarty** [comment]
Thank you!
